test/sql/function/timestamp/test_icu_time_bucket_timestamptz.test

```diff
@@ -1213,3 +1213,40 @@ with t2 as (select '2000-10-28 00:00:00'::timestamptz + (n::text || 'hours')::in
 2000-10-30 17:00:00-08	2000-10-30 17:00:00-08
 2000-10-30 20:00:00-08	2000-10-30 20:00:00-08
 2000-10-30 23:00:00-08	2000-10-30 23:00:00-08
+
```

Comment:
This is great but can we add all the tests from our discussion? I'm really paranoid now.  And can you add some from another TZ?

Reply:
added a couple more -- FWIW i am copy / pasting the output of the queries from timescaledb into the expected output for the test. so far so good.

Reply:
added this one:
```sql
select
  bool_and(
    date_part('epoch', time_bucket(interval '3' day, timestamptz '2023-06-07 16:08:09+00', origin))
    = date_part('epoch', time_bucket(interval '3' day, timestamp   '2023-06-07 16:08:09', origin at time zone 'UTC'))
  )
from
  generate_series(timestamptz '2023-01-03 00:00:00+05',
                  timestamptz '2024-01-04 00:00:00+05',
                  interval '7877' minute) as t(origin)
;
```
which makes sure that the icu and non-icu versions are equal.